spring-boot/admin/src/main/resources/templates/admin/fragments/wizard/wizard.html

```diff
@@ -1,348 +1,381 @@
-                {
-                    alert("You must fill service name");
-                }
-                if ($("#serviceUrl").val().trim() == "")
-                {
-                    alert("You must fill service url");
-                }
-
-                if (!isValidURL($("#serviceUrl").val().trim()))
-                {
-                    alert("Service URL must be valid and start with http:// or https://");
-                }
-
-                if ($("#serviceForFree").val() == "")
-                {
-                    alert("You must select either free service or not");
-                } else
-                {
-                    if ($("#serviceForFree").val() == "false" && $("#monthlyChargeInput").val() == "")
-                    {
-                        alert("You must select either monthly charge or charge per call");
-                    }
-
-                    if ($("#serviceForFree").val() == "false" && $("#serviceCost").val() == "")
-                    {
-                        alert("You must set up a cost for your service");
-                    }
-                }
-
-                if ($("#serviceName").val().trim() != "" && $("#serviceUrl").val().trim() && isValidURL($("#serviceUrl").val().trim()))
-                {
-                    if ($("#serviceForFree").val() !== "")
-                    {
-                        if ($("#serviceForFree").val() == "true")
-                        {
-                            $("#wizardForm").submit();
-                            $('#wizard').hide();
-                            $('#busy_indicator').show();
-                        } else
-                        {
-                            if ($("#monthlyChargeInput").val() !== "" && $("#serviceCost").val() !== "")
-                            {
-                                $("#wizardForm").submit();
-                                $('#wizard').hide();
-                                $('#busy_indicator').show();
-                            }
-                        }
-                    }
-
-                }
+        );
+
+        $('.buttonNext').addClass('btn btn-success');
+        $('.buttonPrevious').addClass('btn btn-primary');
+        $('.buttonFinish').addClass('btn btn-default');
+
+    }
+
+    function leaveAStepCallback(obj, context) {
+        if (context.toStep > context.fromStep)
+            return validateSteps(context.fromStep);
+        else
+            return true;
+    }
+
+    function validateSteps(stepnumber) {
+        //Step 1 and 5 don't need any validation.
+        var isStepTwoValid = validateServiceName(stepnumber);
+        var isStepThreeValid = validateURL(stepnumber);
+        var isStepFourValid = validateServiceCharge(stepnumber);
+
+        return isStepTwoValid && isStepThreeValid && isStepFourValid;
+    }
+
+    function validateServiceName(stepnumber) {
+        if (stepnumber > 1 && $("#serviceName").val().trim() == "") {
```

Comment:
I do not understand why the stepnumber is passed to this function at all. If it is "1" there will be no validation at all. So I would suggest to call this function _only_ if `stepnumber > 1`. Then the parameter can be removed. This applies also to the other validate methods that take a stepnumber as parameter.

Reply:
Done